usr/src/uts/common/fs/zfs/dmu.c

```diff
@@ -1992,6 +1998,12 @@ dmu_sync(zio_t *pio, uint64_t txg, dmu_sync_cb_t *done, zgd_t *zgd)
 	dsa->dsa_zgd = zgd;
 	dsa->dsa_tx = NULL;
 
+	/*
+	 * Record the vdev(s) backing this blkptr so they can be flushed when
+	 * the writes for the lwb have completed.
+	 */
+	zil_lwb_add_block(zgd->zgd_lwb, zgd->zgd_bp);
```

Comment:
same here (well, it's actually a valid BP, but it's the old location of this logical block, used for nopwrite).